megnet/dataloader/MEGNetDataset.py

```diff
@@ -0,0 +1,134 @@
+"""
+Tools to construct a dataloader for DGL grphs
+"""
+import os
+
+import dgl
+import torch
+from dgl.data import DGLDataset
+from dgl.data.utils import load_graphs, save_graphs
+from torch.utils.data import DataLoader
+from tqdm import trange
+
+
+def _collate_fn(batch):
+    """
+    Merge a list of dgl graphs to form a batch
+    """
+    graphs, labels = map(list, zip(*batch))
+    g = dgl.batch(graphs)
+    labels = torch.tensor(labels, dtype=torch.float32)
+    return g, labels
+
+
+def MEGNetDataLoader(train_data, val_data, test_data, collate_fn, batch_size, num_workers):
+    """
+    Dataloader for MEGNet training
+    Args:
+    train_data: training data
+    val_data: validation data
+    test_data: testing data
+    collate_fn:
+    """
+    train_loader = DataLoader(
+        train_data,
+        batch_size=batch_size,
+        shuffle=True,
+        collate_fn=collate_fn,
+        num_workers=num_workers,
+        pin_memory=True,
+    )
+
+    val_loader = DataLoader(
+        val_data,
+        batch_size=batch_size,
+        shuffle=False,
+        collate_fn=collate_fn,
+        num_workers=num_workers,
+        pin_memory=True,
+    )
+
+    test_loader = DataLoader(
+        test_data,
+        batch_size=batch_size,
+        shuffle=False,
+        collate_fn=collate_fn,
+        num_workers=num_workers,
+        pin_memory=True,
+    )
+    return train_loader, val_loader, test_loader
+
+
+class MEGNetDataset(DGLDataset):
+    """
+    Create a dataset including dgl graphs
+    """
+
+    def __init__(self, structures, labels, label_name, cry_graph):
+        """
+        Args:
+        structures: Pymatgen strutcure
+        labels: property values
+        label: label name
+        cry_graph: Pmg2Graph
+        """
+        self.cry_graph = cry_graph
+        self.structures = structures
+        self.labels = torch.FloatTensor(labels)
+        self.label_name = label_name
+        self.path = os.getcwd()
```

Comment:
This is not necessary. All paths are by default from the current working directory. You do not need this variable.